### PR TITLE
bau: Various authorisation code cleanups

### PR DESCRIPTION
**Pull request by oswaldquek**
See individual commits for what has been done here.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/wallets/WalletAuthoriseService.java [hunk 1]
```diff
@@ -52,24 +51,15 @@ public WalletAuthoriseService(PaymentProviders paymentProviders,
...
             ChargeStatus chargeStatus = null;
-            String responseFromPaymentGateway = null;
             String requestStatus = "failure";
 
             try {
                 operationResponse = authorise(charge, walletAuthorisationData);
-                Optional<BaseAuthoriseResponse> baseResponse = operationResponse.getBaseResponse();
 
-                if (baseResponse.isPresent()) {
+                if (operationResponse.getBaseResponse().isPresent()) {
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/wallets/WalletAuthoriseService.java:60]: Can this be re-jigged to use `ifPresentOrElse(…)`?

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks like some nice incremental changes.

File: src/main/java/uk/gov/pay/connector/wallets/WalletAuthoriseService.java (hunk 1 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/wallets/WalletAuthoriseService.java:60]: I don't think this is going to work...
Screenshot 2020-11-13 at 11 39 57

Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/wallets/WalletAuthoriseService.java:60]: I think the way `CardAuthoriseService` does it is nicer:
```
            try {
                operationResponse = authorise(charge, authCardDetails);

                if (operationResponse.getBaseResponse().isEmpty()) {
                    operationResponse.throwGatewayError();
                }

                newStatus = operationResponse.getBaseResponse().get().authoriseStatus().getMappedChargeStatus();
```

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**